src/ast/effects.h

```diff
@@ -204,6 +204,24 @@ struct EffectAnalyzer : public PostWalker<EffectAnalyzer> {
     isAtomic = true;
     if (!ignoreImplicitTraps) implicitTrap = true;
   }
+  void visitAtomicWait(AtomicWait* curr) {
+    readsMemory = true;
+    // AtomicWait doesn't strictly write memory, but it does modify the waiters
+    // list associated with the specified address, which we can think of as a
+    // write.
+    writesMemory = true;
+    isAtomic = true;
+    if (!ignoreImplicitTraps) implicitTrap = true;
```

Comment:
i haven't read the spec, how can these trap? seems odd that wait/wake could trap...